concent_api/core/views.py

```diff
@@ -1220,6 +1225,632 @@ def store_message_and_message_status(
         receive_message_status.save()
 
 
+def store_subtask(
+    task_id:                        str,
+    subtask_id:                     str,
+    provider_public_key:            bytes,
+    requestor_public_key:           bytes,
+    state:                          Subtask.SubtaskState,
+    next_deadline:                  int,
+    task_to_compute:                message.TaskToCompute                = None,
+    report_computed_task:           message.ReportComputedTask           = None,
+    ack_report_computed_task:       message.AckReportComputedTask        = None,
+    reject_report_computed_task:    message.RejectReportComputedTask     = None,
+    subtask_results_accepted:       message.tasks.SubtaskResultsAccepted = None,
+    subtask_results_rejected:       message.tasks.SubtaskResultsRejected = None,
+):
+    """
+    Validates and stores subtask and its data in Subtask table.
+    Stores related messages in StoredMessage table and adds relation to newly created subtask.
+    """
+    assert isinstance(task_id,              str)
+    assert isinstance(subtask_id,           str)
+    assert isinstance(provider_public_key,  bytes)
+    assert isinstance(requestor_public_key, bytes)
+    assert state in Subtask.SubtaskState
+    assert (state in Subtask.ACTIVE_STATES)  == (next_deadline is not None)
+    assert (state in Subtask.PASSIVE_STATES) == (next_deadline is None)
+
+    provider  = Client.objects.get_or_create_full_clean(provider_public_key)
+    requestor = Client.objects.get_or_create_full_clean(requestor_public_key)
+
+    subtask = Subtask(
+        task_id         = task_id,
+        subtask_id      = subtask_id,
+        provider        = provider,
+        requestor       = requestor,
+        state           = state.name,
+        next_deadline   = parse_timestamp_to_utc_datetime(next_deadline),
+    )
+
+    set_subtask_messages(
+        subtask,
+        task_to_compute             = task_to_compute,
+        report_computed_task        = report_computed_task,
+        ack_report_computed_task    = ack_report_computed_task,
+        reject_report_computed_task = reject_report_computed_task,
+        subtask_results_accepted    = subtask_results_accepted,
+        subtask_results_rejected    = subtask_results_rejected,
+    )
+
+    subtask.full_clean()
+    subtask.save()
+
+    logging.log_subtask_stored(
+        task_id,
+        subtask_id,
+        state.name,
+        subtask.provider.public_key,
+        subtask.requestor.public_key,
+        next_deadline,
+    )
+
+    return subtask
+
+
+def verify_file_status(
+    client_public_key: bytes,
+):
+    """
+    Function to verify existence of a file on cluster storage
+    """
+
+    force_get_task_result_list = PendingResponse.objects.filter(
+        subtask__requestor__public_key  = b64encode(client_public_key),
+        subtask__state                  = Subtask.SubtaskState.FORCING_RESULT_TRANSFER.name,  # pylint: disable=no-member
```

Comment:
This function must not depend on the receive queue. Nothing else than `receive/` should depend on it. This is something this refactoring was supposed to eliminate. You should only use information from `Subtask` table here. 

This message will disappear from the queue when requestor receives it. The behavior of this function is supposed to be the same, no matter whether the requestor comes to us for the message or not.

Reply:
Fixed.